examples/shell/shell_common/cmd_ping_responder.cpp

```diff
@@ -0,0 +1,185 @@
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "shell/streamer.h"
+#include <ChipShellCollection.h>
+
+#include <platform/CHIPDeviceLayer.h>
+#include <protocols/echo/Echo.h>
+#include <protocols/secure_channel/PASESession.h>
+#include <shell/shell_core.h>
+#include <support/ErrorStr.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/raw/TCP.h>
+#include <transport/raw/UDP.h>
+
+using chip::Shell::shell_command_t;
+using chip::Shell::streamer_get;
+using chip::Shell::streamer_printf;
+
+namespace {
+
+class EchoServerCmd
+{
+public:
+    EchoServerCmd() = default;
+
+    CHIP_ERROR Start(bool tcp)
+    {
+        if (mConnected)
+        {
+            return CHIP_ERROR_INCORRECT_STATE;
+        }
+        chip::Optional<chip::Transport::PeerAddress> peer(chip::Transport::Type::kUndefined);
+        mAdminInfo = mAdmins.AssignAdminId(kAdminId, chip::kTestDeviceNodeId);
+        if (mAdminInfo == NULL)
+        {
+            return CHIP_ERROR_NO_MEMORY;
+        }
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+        mTcp = tcp;
+        if (tcp)
+        {
+            ReturnErrorOnFailure(mTCPManager.Init(chip::Transport::TcpListenParameters(&chip::DeviceLayer::InetLayer)
+                                                      .SetAddressType(chip::Inet::kIPAddressType_IPv4)));
+            ReturnErrorOnFailure(
+                mSessionManager.Init(chip::kTestDeviceNodeId, &chip::DeviceLayer::SystemLayer, &mTCPManager, &mAdmins));
+        }
+#endif // INET_CONFIG_ENABLE_TCP_ENDPOINT
+        else
+        {
+            ReturnErrorOnFailure(mUDPManager.Init(chip::Transport::UdpListenParameters(&chip::DeviceLayer::InetLayer)
+                                                      .SetAddressType(chip::Inet::kIPAddressType_IPv4)));
+            ReturnErrorOnFailure(
+                mSessionManager.Init(chip::kTestDeviceNodeId, &chip::DeviceLayer::SystemLayer, &mUDPManager, &mAdmins));
+        }
+        ReturnErrorOnFailure(mExchangeManager.Init(&mSessionManager));
+        ReturnErrorOnFailure(mEchoServer.Init(&mExchangeManager));
+
+        ReturnErrorOnFailure(mSessionManager.NewPairing(peer, chip::kTestControllerNodeId, &mTestPairing,
+                                                        chip::SecureSessionMgr::PairingDirection::kResponder, kAdminId));
+        mEchoServer.SetEchoRequestReceived(EchoServerCmd::HandleEchoRequestReceived);
+        mConnected = true;
+        return CHIP_NO_ERROR;
+    }
+
+    CHIP_ERROR Stop()
+    {
+        if (!mConnected)
+        {
+            return CHIP_ERROR_INCORRECT_STATE;
+        }
+        chip::Transport::PeerAddress peer(chip::Transport::Type::kUndefined);
+#if INET_CONFIG_ENABLE_TCP_ENDPOINT
+        if (mTcp)
+        {
+            mTCPManager.Disconnect(peer);
```

Comment:
is TCPManager supposed to accept a disconnection for kUndefined?

Generally it should have a correct type (kTCP) and a valid IP address for deduplication.

having kUndefined/kInvalidIPAddr seems unusual.